### PR TITLE
Refactor error handling

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

- Refactored error handling to always return errors to urfave/cli/v2
- Updated some of the error messages
- Removed `getRequiredOption`, updated command flags adding Required flag accordingly 

## Why?

- streamlines and centralizes error handling, fits unit tests expectation
- getRequiredOption was adding arbitrary limitations that were not enforced by the temporal API, which is highly questionable approach. This was only making `tctl` harder to use because of seeing more error messages for arbitrary limitations, when in fact the API would allow such values

## Checklist

1. Closes 

2. How was this tested:

unit tests (tbu)

3. Any docs updates needed?

### CODE REVIEW

**Review by stevekinney — APPROVED** [review]

File: cli/namespaceCommands.go [hunk 1]
```diff
@@ -41,42 +41,45 @@ import (
...
+			return fmt.Errorf("option %s format is invalid: %s", FlagRetention, err)
 		}
 	}
 
 	var isGlobalNamespace bool
 	if c.IsSet(FlagIsGlobalNamespace) {
 		isGlobalNamespace, err = strconv.ParseBool(c.String(FlagIsGlobalNamespace))
 		if err != nil {
-			ErrorAndExit(fmt.Sprintf("Option %s format is invalid.", FlagIsGlobalNamespace), err)
+			return fmt.Errorf("Option %s format is invalid: %s.", FlagIsGlobalNamespace, err)
```
Comment (alexshtin) [inline cli/namespaceCommands.go:67]:
```suggestion
			return fmt.Errorf("option %s format is invalid: %w.", FlagIsGlobalNamespace, err)
```

Comment (alexshtin) [reply cli/namespaceCommands.go:67]: Please use `%w` everywhere for inner errors.

File: cli/namespaceCommands.go [hunk 2]
```diff
@@ -245,28 +270,30 @@ func UpdateNamespace(c *cli.Context) {
...
 
 // DescribeNamespace updates a namespace
-func DescribeNamespace(c *cli.Context) {
+func DescribeNamespace(c *cli.Context) error {
 	namespace := c.String(FlagNamespace)
 	namespaceID := c.String(FlagNamespaceID)
 
 	if namespaceID == "" && namespace == "" {
-		ErrorAndExit("At least namespace_id or namespace must be provided.", nil)
+		return fmt.Errorf("provide either namespace id or namespace name flag")
```
Comment (alexshtin) [inline cli/namespaceCommands.go:293]:
```suggestion
		return fmt.Errorf("provide either %s or %s flag", FlagNamespaceID, FlagNamespace)
```

File: cli/namespaceCommands.go [hunk 3]
```diff
@@ -245,28 +270,30 @@ func UpdateNamespace(c *cli.Context) {
...
 	}
 	if c.IsSet(FlagNamespace) && namespaceID != "" {
-		ErrorAndExit("Only one of namespace_id or namespace must be provided.", nil)
+		return fmt.Errorf("provide only one of the flags: namespace id or namespace name")
```
Comment (alexshtin) [inline cli/namespaceCommands.go:296]:
```suggestion
		return fmt.Errorf("provide only one of the flags: %s or %s", FlagNamespaceID, FlagNamespace)
```

File: cli/namespaceCommands.go [hunk 4]
```diff
@@ -365,16 +401,16 @@ func clustersToString(clusters []*replicationpb.ClusterReplicationConfig) string
...
 		switch c.String(stateFlagName) {
 		case "disabled":
-			return enumspb.ARCHIVAL_STATE_DISABLED
+			return enumspb.ARCHIVAL_STATE_DISABLED, nil
 		case "enabled":
-			return enumspb.ARCHIVAL_STATE_ENABLED
+			return enumspb.ARCHIVAL_STATE_ENABLED, nil
 		default:
-			ErrorAndExit(fmt.Sprintf("Option %s format is invalid.", stateFlagName), errors.New("invalid state, valid values are \"disabled\" and \"enabled\""))
+			return 0, fmt.Errorf("option %s format is invalid: %s", stateFlagName, errors.New("invalid state, valid values are \"disabled\" and \"enabled\""))
```
Comment (alexshtin) [inline cli/namespaceCommands.go:412]:
```suggestion
			return 0, fmt.Errorf("option %s format is invalid: invalid state, valid values are \"disabled\" and \"enabled\"", stateFlagName)
```

**Review by alexshtin — APPROVED** [review]

**Pull request merged**